WinCairo doesn't build without USE_GRAPHICS_LAYER_WC enabled

donny-dont opened the pull request:
#### f26d171c4cc62f4c5fdb3a80be75d7c452eb80e6
<pre>
WinCairo doesn&apos;t build without USE_GRAPHICS_LAYER_WC enabled
<a href="https://bugs.webkit.org/show_bug.cgi?id=245196">https://bugs.webkit.org/show_bug.cgi?id=245196</a>

Reviewed by Fujii Hironori.

Fix the build when `USE_GRAPHICS_LAYER_WC` is disabled by adding checks
around WC code.

* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:
* Source/WebKit/UIProcess/win/PageClientImpl.cpp:
* Source/WebKit/UIProcess/win/PageClientImpl.h:
* Source/WebKit/UIProcess/win/WebView.cpp:
* Source/WebKit/WebProcess/GPU/graphics/wc/WCPlatformLayerGCGL.h:
* Source/WebKit/WebProcess/WebPage/wc/DrawingAreaWC.cpp:
* Source/WebKit/WebProcess/WebPage/wc/DrawingAreaWC.h:
* Source/WebKit/WebProcess/WebPage/wc/GraphicsLayerWC.cpp:
* Source/WebKit/WebProcess/WebPage/wc/GraphicsLayerWC.h:
* Source/WebKit/WebProcess/WebPage/wc/WCBackingStore.h:
* Source/WebKit/WebProcess/WebPage/wc/WCLayerFactory.cpp:
* Source/WebKit/WebProcess/WebPage/wc/WCLayerFactory.h:
* Source/WebKit/WebProcess/WebPage/wc/WCTileGrid.cpp:
* Source/WebKit/WebProcess/WebPage/wc/WCTileGrid.h:
* Source/WebKit/WebProcess/WebPage/wc/WCUpateInfo.h:

Canonical link: <a href="https://commits.webkit.org/254503@main">https://commits.webkit.org/254503@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bf6d90071761322e36b36b74227b6fbc0d8459b6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89236 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33797 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20044 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98550 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32303 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/27840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81595 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93014 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25657 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76148 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25592 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80523 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/80489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68570 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30080 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29805 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15512 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3160 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33252 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38473 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31955 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34603 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
Honestly speaking, I don't like this change because WinCairo doesn't plan to disable it . But, I don't object to land this change if you want.

donny-dont committed:
WinCairo doesn't build without USE_GRAPHICS_LAYER_WC enabled
https://bugs.webkit.org/show_bug.cgi?id=245196

Reviewed by Fujii Hironori.

Fix the build when `USE_GRAPHICS_LAYER_WC` is disabled by adding checks
around WC code.

* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:
* Source/WebKit/UIProcess/win/PageClientImpl.cpp:
* Source/WebKit/UIProcess/win/PageClientImpl.h:
* Source/WebKit/UIProcess/win/WebView.cpp:
* Source/WebKit/WebProcess/GPU/graphics/wc/WCPlatformLayerGCGL.h:
* Source/WebKit/WebProcess/WebPage/wc/DrawingAreaWC.cpp:
* Source/WebKit/WebProcess/WebPage/wc/DrawingAreaWC.h:
* Source/WebKit/WebProcess/WebPage/wc/GraphicsLayerWC.cpp:
* Source/WebKit/WebProcess/WebPage/wc/GraphicsLayerWC.h:
* Source/WebKit/WebProcess/WebPage/wc/WCBackingStore.h:
* Source/WebKit/WebProcess/WebPage/wc/WCLayerFactory.cpp:
* Source/WebKit/WebProcess/WebPage/wc/WCLayerFactory.h:
* Source/WebKit/WebProcess/WebPage/wc/WCTileGrid.cpp:
* Source/WebKit/WebProcess/WebPage/wc/WCTileGrid.h:
* Source/WebKit/WebProcess/WebPage/wc/WCUpateInfo.h:

Canonical link: https://commits.webkit.org/254503@main
